common/model-views.cpp

```diff
@@ -2626,95 +2662,119 @@ namespace rs2
         show_icon(font_18, "recording_icon", u8"\uf111", x, y, id, from_rgba(255, 46, 54, alpha_delta * 255));
     }
 
-    void async_pointclound_mapper::proccess(rs2::frame f, const rs2::frame_source& source)
+    rs2::frame post_processing_filters::apply_filters(rs2::frame f)
     {
 
-        points p;
-        std::vector<frame> results;
-
-
-        if(viewer.synchronization_enable)
+        if (f.get_profile().stream_type() == RS2_STREAM_DEPTH)
         {
-            if (auto composite = f.as<rs2::frameset>())
+            //std::lock_guard<std::mutex> lock(streams_mutex);
+            for (auto&& s : viewer.streams)
             {
-                for(auto&& f: composite)
-                {
-                    if(f.get_profile().unique_id() == viewer.selected_depth_source_uid)
-                    {
-                        p = pc.calculate(f);
-                        results.push_back(std::move(p));
+                if(!s.second.dev) continue;
+                auto dev = s.second.dev;
+                //lock.unlock();
 
-                    }
-                    if(f.get_profile().unique_id() == viewer.selected_tex_source_uid)
+                if (dev->post_processing_enabled)
+                    for (auto&& p : dev->profiles)
                     {
-                        update_texture(f);
-                        results.push_back(std::move(f));
-                    }
-                }
-            }
+                        if (p.stream_type() == RS2_STREAM_DEPTH )
+                        {
+                            auto dec_filter = s.second.dev->decimation_filter;
```

Comment:
I would have prefered to see this declaration and the usage together:
```cpp
if (auto dec_filter = s.second.dev->decimation_filter)
    if(dec_filter->enabled)
        f = dec_filter->invoke(f);
```

Reply:
same for other variables

Reply:
Agree